src/lib.rs

```diff
@@ -0,0 +1,1167 @@
+    /// bits.
+    ///
+    /// # Panics
+    ///
+    /// If the two Vobs are of different length.
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use] extern crate vob;
+    /// fn main() {
+    ///     let mut v1 = vob![true, false, true];
+    ///     let v2 = vob![false, true, true];
+    ///     v1.xor(&v2);
+    ///     assert_eq!(v1, vob![true, true, false]);
+    /// }
+    /// ```
+    pub fn xor(&mut self, other: &Vob<T>) -> bool {
+        if self.bits != other.bits {
+            panic!("Cannot 'xor' two Vobs of different length ({}  {})", self.bits, other.bits);
+        }
+        let mut chngd = false;
+        for (sb, ob) in self.vec.iter_mut().zip(other.vec.iter()) {
+            let ov = *sb;
+            let nv = ov ^ *ob;
+            if ov != nv {
+                chngd = true;
+            }
+            *sb = nv;
+        }
+        self.mask_last_block();
+        chngd
+    }
+
+    /// We guarantee that the last storage block has no bits set past the "last" bit: this function
+    /// clears any such bits.
+    fn mask_last_block(&mut self) {
+        let db = self.bits % bits_per_block::<T>();
+        if db > 0 {
+            let m = (T::one() << db) - T::one();
+            let b = storage_block_off::<T>(self.bits);
+            let v = self.vec[b];
+            self.vec[b] = v & m;
+        }
+    }
+}
+
+impl Default for Vob<usize> {
+    fn default() -> Self {
+        Vob {
+            bits: 0,
+            vec: Vec::new()
+        }
+    }
+}
+
+impl<T: Debug + One + PrimInt + Zero> Debug for Vob<T> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "Vob[")?;
+        for b in self {
+            write!(fmt, "{}", if b { 1 } else { 0 })?;
+        }
+        write!(fmt, "]")?;
+        Ok(())
+    }
+}
+
+impl<T: Debug + One + PrimInt + Zero> Extend<bool> for Vob<T> {
+    fn extend<I: IntoIterator<Item=bool>>(&mut self, iterable: I) {
+        let iterator = iterable.into_iter();
+        let (min, _) = iterator.size_hint();
+        self.reserve(min);
+        for e in iterator {
+            self.push(e)
+        }
+    }
+}
+
+impl FromIterator<bool> for Vob<usize> {
+    /// Create a Vob from an iterator.
+    ///
+    /// # Examples
+    /// ```
+    /// use std::iter::FromIterator;
+    /// use vob::Vob;
+    /// let v = Vob::from_iter(vec![true, false]);
+    /// assert_eq!(v, Vob::from_iter(vec![true, false]));
+    /// ```
+    fn from_iter<I: IntoIterator<Item=bool>>(iter: I) -> Self {
+        let mut v = Vob::new();
+        v.extend(iter);
+        v
+    }
+}
+
+
+// This is based on the `BitVec` approach to indices. It's clearly a horrible way of doing things,
+// but until `IndexGet` is implemented, we're stuck.
+static TRUE: bool = true;
+static FALSE: bool = false;
+
+impl<T: Debug + One + PrimInt + Zero> Index<usize> for Vob<T> {
+    type Output = bool;
+
+    fn index(&self, index: usize) -> &bool {
```

Comment:
Hah. So a reference received by this method can't observe a mutation?

Reply:
?

Reply:
Yeah, I think I'm talking garbage too.

There's no way to mutate a vector without going via a mutable reference, is there? Even if you own the vec.

Reply:
Correct.

Reply:
Right, so disregard my comment :)

Reply:
The problem I had in my head would only manifest if you could obtain a mutable reference to a bit in side a `Vob`. Do you see what I mean?

We don't support that, so it's moot.

You'd have to use a separate struct if you wanted such an interface:
```
struct MutBit<'v> {
    vob: &mut 'v Vob,
    bit: usize,
}

impl MutBit {
  fn set(self, val: bool) {
        ....
  }
}
```

Reply:
Indeed, it's more than I want to support now.